### PR TITLE
Use event delegation in phoenix_html.js #75

### PR DESCRIPTION
**Pull request by Troush**
Refers to #75

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you @Troush! Maybe we should drop support for IE8? Folks can write their own tiny phoenix-html script if they need to support older browsers.

File: priv/static/phoenix_html.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+function matches(elm, selector) {
+    var matches = (elm.document || elm.ownerDocument).querySelectorAll(selector),
+        i = matches.length;
+    while (--i >= 0 && matches.item(i) !== elm) {}
+    return i > -1;
 }
 
-;
\ No newline at end of file
+window.addEventListener('click', function (event) {
```
Comment (chrismccord) [inline priv/static/phoenix_html.js]: two space indentation please

File: priv/static/phoenix_html.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+	// Although ^=parent is not technically correct,
+	// we need to use it in order to get IE8 support.
+	if(event.target && matches(event.target, 'a[data-submit^=parent]')) {
+	    var message = event.target.getAttribute("data-confirm");
```
Comment (chrismccord) [inline priv/static/phoenix_html.js]: use consistent quoting, either use is fine

**Comment by Troush** [comment]
Updated. Without IE8 support, and indentation fixes.
Sorry for indentation :\

**Comment by chrismccord** [comment]
<3 <3 <3

**Comment by maxgronlund** [comment]
Worked great for a short while. how it's broken, keep it up

**Comment by josevalim** [comment]
@maxgronlund I think there is a typo in your message and that made me confused. :) It is working now or is it broken? :)

**Comment by Troush** [comment]
@maxgronlund Describe more detail whats working wrong?

I will happy to fix if something wrong :)

**Comment by jsteiner** [comment]
It seems this change breaks compatibility with `phantomjs` (`2.1.1`, the latest), as `phantomjs` doesn't support `Element.matches`. Adding a polyfill fixed it locally.

Would a fine solution here be to update`phoenix_html` to include the polyfill?

**Comment by josevalim** [comment]
@jsteiner instead of introducing matches, what if we explicitly check for the tag and the attribute of the element since that's the only information we care about? I think it is going to be simpler/faster than including the polyfill? Could you please send a PR?

**Pull request merged**